### PR TITLE
Make ones and zeros's ref accepts variadic size argument

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85117

### CODE REVIEW

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -3404,14 +3404,15 @@ def new_empty_strided(
 @register_decomposition(torch.ops.aten.zeros)
 @out_wrapper()
 def zeros(
-    size: ShapeType,
-    *,
```
Comment (ngimel) [inline torch/_refs/__init__.py:3408]: why did you remove kwarg-only? dtype and following args are kwarg only.

Comment (SherlockNoMad) [reply torch/_refs/__init__.py:3408]: Python syntax actually doesn't allow this:  Keyword-only argument separator not allowed after "*parameter"

Coz, arguments after "*parameter" are implicitly kwarg-only...

Comment (SherlockNoMad) [reply torch/_refs/__init__.py:3408]:
```
@overload
def ones(size: _size, *, out: Optional[Tensor]=None, dtype: Optional[_dtype]=None, layout: Optional[_layout]=None, device: Optional[Union[_device, str, None]]=None, pin_memory: Optional[_bool]=False, requires_grad: Optional[_bool]=False) -> Tensor: ...
@overload
def ones(*size: _int, out: Optional[Tensor]=None, dtype: Optional[_dtype]=None, layout: Optional[_layout]=None, device: Optional[Union[_device, str, None]]=None, pin_memory: Optional[_bool]=False, requires_grad: Optional[_bool]=False) -> Tensor: ...
```

These are the overloads for torch.ones. 
So def(*size, *, ...) is actually invalid

Comment (ngimel) [reply torch/_refs/__init__.py:3408]: ok cool, can you add some sample or reference input to exercise new code path? Apparently tests were not failing previously even though they should have.

Comment (SherlockNoMad) [reply torch/_refs/__init__.py:3408]: I added the test cases in opinf db for ones and zeros....
However, the test infra is alway normalizing the inputs to a tuple, so technically it's still not testing the non-tuple size case... 
I guess if we really want to test the non-tuple case, we need to carve out an code path just for this purpose. Do we really want to do this?

Comment (ngimel) [reply torch/_refs/__init__.py:3408]: Yeah ok, we can leave it as is.

**Review by ngimel — APPROVED** [review]

**Review by lezcano — APPROVED** [review]
Cool, thank you!
